File: frontend/desktop/lib/models/enums/city.dart
/// Mirrors the backend's `eTicketing.Contracts.Persistence.City` ordinals exactly (Sarajevo(0),
/// Mostar(1), BanjaLuka(2), Tuzla(3), Zenica(4), Bihac(5), Brcko(6), Trebinje(7) — see
/// .claude/rules/01-domain.md). Fixed set of BiH cities a Product can be located in; raw-numeric
/// JSON convention, same as TicketingMode/RoleType.
enum City {
  sarajevo(0),
  mostar(1),
  banjaLuka(2),
  tuzla(3),
  zenica(4),
  bihac(5),
  brcko(6),
  trebinje(7);

  final int value;
  const City(this.value);

  static City fromValue(int value) => City.values.firstWhere(
        (c) => c.value == value,
        orElse: () => City.sarajevo,
      );

  /// Bosnian label for UI display (dropdowns).
  String get label => switch (this) {
        City.sarajevo => 'Sarajevo',
        City.mostar => 'Mostar',
        City.banjaLuka => 'Banja Luka',
        City.tuzla => 'Tuzla',
        City.zenica => 'Zenica',
        City.bihac => 'Bihać',
        City.brcko => 'Brčko',
        City.trebinje => 'Trebinje',
      };
}

File: backend/shared/eTicketing.Contracts/Persistence/City.cs
namespace eTicketing.Contracts.Persistence;

/// <summary>
/// Fixed set of Bosnia and Herzegovina cities a Product can be located in — drives the
/// server-side location filter on GET /products* and the city picker shown when an organizer
/// creates a Product. Declaration order is the wire contract: every frontend (web, desktop,
/// mobile) mirrors this exact order in its own enum, since the value crosses the wire as an
/// ordinal. Extensible later by appending new values — never reorder or remove existing ones.
/// </summary>
public enum City
{
    Sarajevo,
    Mostar,
    BanjaLuka,
    Tuzla,
    Zenica,
    Bihac,
    Brcko,
    Trebinje
}
